app/domain/authentication/resource_restrictions/get_restriction_from_annotation.rb

```diff
@@ -0,0 +1,44 @@
+require 'command_class'
+
+module Authentication
+  module ResourceRestrictions
+    # Parses the given annotation name and tries to extract a restriction from it.
+    # A restriction can have 2 formats:
+    # 1. "<Authenticator name>/<Restriction name>"
+    # 2. "<Authenticator name>/<Service ID>/<Restriction name>"
+    #
+    # The first format is a general restriction.
+    # This means it is relevant to all services of this authenticator.
+    #
+    # The second format is specific to the specified service ID.
+    # This means all other services will ignore it.
+    # Moreover, if both formats are found for the same restriction, the specific one will be used.
+    #
+    # This function returns the restriction name, if found (nil otherwise) and a
+    # boolean indicating if it is a general restriction or not.
+    GetRestrictionFromAnnotation = CommandClass.new(
+      dependencies: {},
+      inputs: %i[annotation_name authenticator_name service_id]
+    ) do
```

Comment:
add blank line between `do` and `def call`

Reply:
Done